core/src/display_object/movie_clip.rs

```diff
@@ -1711,6 +1715,10 @@ impl<'gc> TDisplayObject<'gc> for MovieClip<'gc> {
                 };
                 let _ = tag_utils::decode_tags(&mut reader, tag_callback, TagCode::ShowFrame);
             }
+
+            if !self.0.read().avm2_constructor_ran {
+                self.construct_as_avm2_object(context);
+            }
```

Comment:
Could we instead have set a local bool in the `if matches!(self.object2(), Avm2Value::Undefined) {` line above to indicate that we need construction? That way we don't need to store the extra flag inside MovieClip.

Somewhat separately, but there's also a difference with EditText/Graphic where they do the AVM2 object construction in `post_instantiation` still -- not a big deal there since they don't really have children, but I figure it should probably be consistent and have this code inside `construct_frame`.